Sources/Operators/ZipMany.swift

```diff
@@ -42,31 +42,51 @@ public extension Collection where Element: Publisher {
     func zip() -> AnyPublisher<[Element.Output], Element.Failure> {
         if isEmpty { return Empty().eraseToAnyPublisher() }
 
-        // Given an array of publishers, return an array of the result of zip'ing each successive foursome of
-        // input publishers (or fewer, for the remainder of count%4).
-        func makeZippedQuads(input: [AnyPublisher<[Element.Output], Element.Failure>]) -> [AnyPublisher<[Element.Output], Element.Failure>] {
-            sequence(state: input.makeIterator(), next: { it in it.next().map { ($0, it.next(), it.next(), it.next()) } })
-            .map { quad in
-                guard let second = quad.1 else { return quad.0 }
-                guard let third = quad.2 else { return quad.0.zip(second)
-                        .map { $0.0 + $0.1 }
-                        .eraseToAnyPublisher()
-                }
-                guard let fourth = quad.3 else { return quad.0.zip(second, third)
-                        .map { $0.0 + $0.1 + $0.2 }
-                        .eraseToAnyPublisher()
-                }
-                return quad.0.zip(second, third, fourth)
-                    .map { $0.0 + $0.1 + $0.2 + $0.3 }
-                    .eraseToAnyPublisher()
-            }
-        }
-
         var wrapped = map { $0.map { [$0] }.eraseToAnyPublisher() }
         while wrapped.count > 1 {
             wrapped = makeZippedQuads(input: wrapped)
         }
-        return wrapped[0].eraseToAnyPublisher()
+        return wrapped.first?.eraseToAnyPublisher() ?? Empty().eraseToAnyPublisher()
```

Comment:
ditto

Reply:
Good point, mind removing the lines at 49 and I'll merge it?

Reply:
Done.